reporter/src/main/kotlin/reporters/AmazonOssAttributionBuilderReporter.kt

```diff
@@ -0,0 +1,199 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.reporter.reporters
+
+import java.io.IOException
+import java.io.OutputStream
+import java.net.ConnectException
+import java.time.Instant
+
+import okhttp3.Credentials
+
+import org.ossreviewtoolkit.model.jsonMapper
+import org.ossreviewtoolkit.model.utils.getDetectedLicensesWithCopyrights
+import org.ossreviewtoolkit.reporter.HTTP_CACHE_PATH
+import org.ossreviewtoolkit.reporter.Reporter
+import org.ossreviewtoolkit.reporter.ReporterInput
+import org.ossreviewtoolkit.reporter.utils.AmazonOssAttributionBuilderService
+import org.ossreviewtoolkit.utils.OkHttpClientHelper
+import org.ossreviewtoolkit.utils.log
+import org.ossreviewtoolkit.utils.showStackTrace
+
+import retrofit2.Response
+
+private fun <T> getBodyOrHandleError(response: Response<T>, errorTitle: String): T {
+    val body = response.body()
+    if (response.isSuccessful && body != null) return body
+
+    val errorMessage = response.errorBody()?.let { errorBody ->
+        val errorResponse = jsonMapper.readValue(
+            errorBody.string(),
+            AmazonOssAttributionBuilderService.ErrorResponse::class.java
+        )
+
+        errorResponse.error
+    } ?: "Error code ${response.code()}"
+
+    throw IOException("$errorTitle: $errorMessage")
+}
+
+/**
+ * A [Reporter] that creates an attribution document in .txt format using a running instance of the Amazon Attribution
+ * Builder (https://github.com/amzn/oss-attribution-builder).
+ */
+class AmazonOssAttributionBuilderReporter : Reporter {
+    override val reporterName = "AmazonOssAttributionBuilder"
+    override val defaultFilename = "OssAttribution.txt"
+
+    private val service = AmazonOssAttributionBuilderService.create(
+        AmazonOssAttributionBuilderService.Server.DEFAULT,
+        OkHttpClientHelper.buildClient(HTTP_CACHE_PATH)
+    )
+
+    override fun generateReport(outputStream: OutputStream, input: ReporterInput) {
+        // TODO: Allow to configure username and password via the config file; use the defaults for now.
+        val credentials = Credentials.basic("admin", "admin")
+
+        val rootProject = input.ortResult.getProjects().singleOrNull()
+
+        // TODO: Add support for multiple projects.
+        requireNotNull(rootProject) {
+            "The AmazonOssAttributionBuilderReporter currently only supports ORT results with a single project"
```

Comment:
How about using `$reporterName` instead of hard-coding "AmazonOssAttributionBuilderReporter"? And please keep the dot at the end of the sentence.